### PR TITLE
Optional catch-all routes

### PR DESCRIPTION
**Pull request by Vadorequest**
https://github.com/vercel/next.js/discussions/16907

### CODE REVIEW

**Review by samuelcastro — APPROVED** [review]
Looks great, are you planning on merge this in?

**Review by samuelcastro — DISMISSED** [review]

**Comment by Vadorequest** [comment]
@samuelcastro Not planning on merging it because of https://github.com/vercel/next.js/discussions/16907

Basically, it doesn't work. Theoretically everything is fine, but in practices I feel like I missed something, or it's not meant for SSG (which isn't obvious).

I had completely forgotten about this PR to be honest, there is quite a great deal of documentation that should be merged nevertheless. I just pinged Timer on the issue and hope I'll get some feedback soon.

**Comment by samuelcastro** [comment]
It seems you just need to use fallback: true

**Comment by Vadorequest** [comment]
Yep, saw that yesterday too. Need to try it.

File: package.json [hunk 1]
```diff
@@ -102,7 +103,7 @@
     "lodash.startswith": "4.2.1",
     "lodash.xorby": "4.7.0",
     "markdown-to-jsx": "6.11.4",
-    "next": "9.4.4",
+    "next": "9.5.4-canary.4",
```
Comment (Vadorequest) [inline package.json]: Need to get out of canary

File: src/pages/[locale]/examples/native-features/example-optional-catch-all-routes/[[...slug]].tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    return {
+      params: {
+        locale: supportedLocale.name,
+        slug: [],
+      },
+    };
+  });
+
+  return {
+    fallback: false,
```
Comment (Vadorequest) [inline src/pages/[locale]/examples/native-features/example-optional-catch-all-routes/[[...slug]].tsx]: should be true 

See https://github.com/vercel/next.js/discussions/16907

File: package.json (hunk 1 above)
Comment (Vadorequest) [reply package.json]: This can be done later.

**Comment by Vadorequest** [comment]
@samuelcastro Just tested and it does work! I'll finish the documentation example and merge this ASAP.

**Comment by samuelcastro** [comment]
Could you rebase those changes into `v1-ssr-mst-aptd-gcms-lcz-sty` as well?

**Comment by Vadorequest** [comment]
@samuelcastro It's done, I was actually working on it 😉

**Comment by Vadorequest** [comment]
This merge was a mistake and has been wiped from the git history, https://github.com/UnlyEd/next-right-now/pull/173 was squashed & merged instead.

I either selected the "merge" option by mistake, or github played me a sad trick. (ran into an issue while merging)

**Pull request merged**